### ISSUE TITLE
org.apache.httpcomponents.httpclient referenced with two different versions

### ISSUE THREAD

**Issue by dominicdesu** [eclipse/smarthome#678]
Throughout the ESH project, it seems that two different versions of org.apache.httpcomponents.httpclient are referenced:

In the selected_target_plugins of [SmartHome Runtime.launch](https://github.com/eclipse/smarthome/blob/876d10e6aa20aeaa2dc61ae72354aac663e64336/targetplatform/SmartHome%20Runtime.launch#L25), two different versions are referenced:
`org.apache.httpcomponents.httpclient*4.2.6.v201311072007@default:default,org.apache.httpcomponents.httpclient*4.3.6.v201411290715@default:default`

The [smarthome.target](https://github.com/eclipse/smarthome/blob/876d10e6aa20aeaa2dc61ae72354aac663e64336/targetplatform/smarthome.target#L45) only references the "old" version:
`<unit id="org.apache.httpcomponents.httpclient" version="4.2.6.v201311072007"/>`

while most of the test launch files (e.g. for [io.rest.test](https://github.com/eclipse/smarthome/blob/876d10e6aa20aeaa2dc61ae72354aac663e64336/bundles/io/org.eclipse.smarthome.io.rest.test/org.eclipse.smarthome.io.rest.test.launch#L37)) reference the "new" version:
`org.apache.httpcomponents.httpclient*4.3.6.v201411290715@default:default`

If no specific version is referenced, as it is currently in the [hue.test launch file](https://github.com/eclipse/smarthome/blob/876d10e6aa20aeaa2dc61ae72354aac663e64336/extensions/binding/org.eclipse.smarthome.binding.hue.test/org.eclipse.smarthome.binding.hue.test.launch#L36) (and in the archetype.binding.test package), the following error is shown when trying to run the launch file:

```
!ENTRY org.eclipse.smarthome.binding.hue 4 0 2015-12-08 12:58:21.362
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: org.eclipse.smarthome.binding.hue [4]
  Unresolved requirement: Import-Package: org.jupnp.model.meta
    -> Export-Package: org.jupnp.model.meta; bundle-symbolic-name="org.jupnp"; bundle-version="2.0.0.201511241047"; version="0.0.0"
       org.jupnp [59]
         Unresolved requirement: Import-Package: org.apache.http.client
           -> Export-Package: org.apache.http.client; bundle-symbolic-name="org.apache.httpcomponents.httpclient"; bundle-version="4.3.6.v201411290715"; version="4.3.6"; uses:="org.apache.http.protocol,  org.apache.http.params,  org.apache.http.conn.routing,  org.apache.http.conn,  org.apache.http.cookie,  org.apache.http.client.methods,  org.apache.http,  org.apache.http.auth"
              org.apache.httpcomponents.httpclient [62]
                Unresolved requirement: Import-Package: net.sf.ehcache; resolution:="optional"
                Unresolved requirement: Import-Package: net.spy.memcached; resolution:="optional"
                Unresolved requirement: Import-Package: org.apache.http; version="4.3.3"
```

Are these two different versions used on purpose?
Probably either the launch files should be adjusted to use the new one, or the old library should not be used at all?

**Comment by maggu2810** [comment]
I don't think this has been done by intention.
IMHO the code should works with both versions (I am currently using 4.2 artifacts myself and the stuff I am using is working).
But the used / referenced / declared version should be the same for the whole ESH tree.

**Comment by kaikreuzer** [comment]
If the target platform defines only version 4.2.6, how doe the version 4.3.6 get in there as well? Does anybody now? I think the best solution would be to only have one version in the target, then we could be sure that all code is referring to the same one.

**Comment by dominicdesu** [comment]
I don't have much experience with OSGI, but is it possible that the 4.3.6 version is somehow imported/used by the [MANIFEST.MF of the jUPnP library](https://github.com/jupnp/jupnp/blob/master/bundles/org.jupnp.osgi/META-INF/MANIFEST.MF#L9)?

**Comment by itheiss** [comment]
Isn't the issue obsolete with the migration to Jetty and can be closed?

**Comment by kaikreuzer** [comment]
Not really as the target platform still contains those two versions.
And we indeed still depend on apache http through jUPnP, so we cannot remove it. I have just created https://github.com/jupnp/jupnp/issues/70; once this is implemented, I think we should be fine to fully remove apache http from the target platform. (And regarding the original question of the issue: No, I still do not know why we have two versions there...)
